test/units/modules/network/dcnm/test_dcnm_vrf.py

```diff
@@ -22,79 +22,274 @@
+        set_module_args(dict(_ansible_check_mode=True, state='merged',
+                             fabric='test_fabric', config=self.playbook_config))
+        self.execute_module(changed=True, failed=False)
 
-    def test_dcnm_vrf_create_duplicate(self):
-        set_module_args(dict(action='create', fabric='test_fabric', vrf_name='test_vrf', vrf_id=919191))
-        result = self.execute_module(changed=False, failed=False)
-        self.assertFalse(result.get('changed'))
+    def test_04dcnm_vrf_merged_new(self):
+        set_module_args(dict(state='merged', fabric='test_fabric', config=self.playbook_config))
+        self.execute_module(changed=True, failed=False)
 
-    def test_dcnm_vrf_create_with_used_vrfid(self):
-        set_module_args(dict(action='create', fabric='test_fabric', vrf_name='test_vrf', vrf_id=919191))
+    def test_05dcnm_vrf_merged_duplicate(self):
+        set_module_args(dict(state='merged', fabric='test_fabric', config=self.playbook_config))
+        self.execute_module(changed=False, failed=False)
+
+    def test_06dcnm_vrf_merged_with_incorrect_vrfid(self):
+        set_module_args(dict(state='merged', fabric='test_fabric', config=self.playbook_config_incorrect_vrfid))
         result = self.execute_module(changed=False, failed=True)
-        self.assertEqual(result.get('msg')[1], 'VRF ID is already in use')
-
-    def test_dcnm_vrf_attach_single_switch(self):
-        set_module_args(dict(action='attach', fabric='test_fabric', vrf_name='test_vrf',
-                             deployment='False', serial_numbers_vlans=[{'947BSAMCAU1':'103'}]))
-        result = self.execute_module(changed=True, failed=False)
-        self.assertTrue(result.get('changed'))
-
-    def test_dcnm_vrf_attach_multiple_switches(self):
-        set_module_args(dict(action='attach', fabric='test_fabric', vrf_name='test_vrf',
-                             deployment='False', serial_numbers_vlans=[{'947BSAMCAU1':'103'},
-                                                                       {'947BSAMCBV1':'104'}]))
-        result = self.execute_module(changed=True, failed=False)
-        self.assertTrue(result.get('changed'))
-
-    def test_dcnm_vrf_attach_with_used_vlan(self):
-        set_module_args(dict(action='attach', fabric='test_fabric', vrf_name='test_vrf',
-                             deployment='False', serial_numbers_vlans=[{'947BSAMCAU1':'103'}]))
+        self.assertEqual(result.get('msg'), 'vrf_id for VRF:test_vrf_1 cant be updated to a different value')
+
+    def test_07dcnm_vrf_merged_with_update(self):
+        set_module_args(dict(state='merged', fabric='test_fabric', config=self.playbook_config_update))
+        self.execute_module(changed=True, failed=False)
+
+    def test_08dcnm_vrf_merged_with_update_vlan(self):
+        set_module_args(dict(state='merged', fabric='test_fabric', config=self.playbook_config_update_vlan))
+        self.execute_module(changed=True, failed=False)
```

Comment:
Maybe I am missing something but I don't see where you are validating any of the return data from the module call in many of these tests?